Tests/RxSwift/ZipWithTest.swift

```diff
@@ -0,0 +1,101 @@
+//
+// Created by Arjan Duijzer on 26/12/2017.
+// Copyright (c) 2017 RxSwiftCommunity. All rights reserved.
+//
+
+import XCTest
+import RxSwift
+import RxTest
+
+struct Pair<F: Equatable, S: Equatable> {
+    let first: F
+    let second: S
+}
+
+enum ZipWithTestError: Error {
+    case error
+}
+
+extension Pair: Equatable {
+}
+
+func ==<F, S>(lhs: Pair<F, S>, rhs: Pair<F, S>) -> Bool {
+    return lhs.first == rhs.first && lhs.second == rhs.second
+}
+
+class ZipWithTest: XCTestCase {
+    func testZipWith_SourcesNotEmpty_ZipCompletes() {
+        let scheduler = TestScheduler(initialClock: 0)
+        let source1 = Observable.just(1)
```

Comment:
I'd personally like the tests to have more than a single pair, and preferable of different types.   (or at least add another test) 

```swift
let source1 = Observable.of(1, 2, 3, 4, 5)
let source2 = Observable.of("a", "b", "c", "d", "e")
```

Reply:
Also, we should have a test for an uneven pair (source1 has 2 items, source2 has 3 items, for example)

Reply:
updated